Marlin/stepper.cpp

```diff
@@ -459,6 +474,10 @@ void Stepper::isr() {
       // #if ENABLED(ADVANCE)
       //   e_steps[TOOL_E_INDEX] = 0;
       // #endif
+
+      #if ENABLED(MK2_MULTIPLEXER)
+        select_multiplexed_stepper(current_block->active_extruder);
```

Comment:
While that is a feasible solution, it just seems to me that the stepper multiplexer should be handled at the lowest level possible in order to allow the higher levels to be agnostic of the physical hardware. Isn't that the whole point of the HAL we're working on for 32-bit?

Regarding the overhead of handling the multiplexer every ISR, I've added a simple check to see if the selected extruder has changed, which means that the only overhead when not changing the multiplexer is a compare and jump, which seems pretty reasonable to me.